test/topology_raft_disabled/test_raft_upgrade.py

```diff
@@ -0,0 +1,94 @@
+#
+# Copyright (C) 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import asyncio
+import pytest
+import logging
+import time
+
+from cassandra.cluster import NoHostAvailable
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.random_tables import RandomTables
+
+logger = logging.getLogger(__name__)
+
+@pytest.mark.asyncio
+async def test_raft_upgrade_basic(manager: ManagerClient, random_tables: RandomTables):
+    start = time.time()
+    servers = await manager.servers()
+
+    cql = manager.cql
+    assert(cql)
```

Comment:
Why is that assert needed? Why doesn't the above code fail already if cql is not available?

Reply:
It's forward protecting the code because it's an `Optional` type.

Reply:
We should probably have a `cql` fixture which does the assert.

Reply:
Why can't "manager.cql" (or "manager.cql()" for less Python magic) assert that there is a CQL connection? What's the point of doing `cql = manager.cql` and then continuing with cql=None? Do we ever want this?

Reply:
`manager.cql` is optional because it may shutdown and reconnect the driver during tests, so there are points in time where the `cql` field is `None`.

Reply:
If we made `cql` into a getter or a property then I guess we could assert it in the getter/property indeed.